File: src/frontend/src/api/schemas/spec.ts
import { z } from 'zod';

export const SpecStatus = z.enum(['Draft', 'InReview', 'Approved', 'Rejected', 'Archived']);
export type SpecStatus = z.infer<typeof SpecStatus>;

export const SpecVersionSchema = z.object({
  id: z.string().uuid(),
  version: z.number().int(),
  content: z.string(),
  createdBy: z.string().nullable(),
  createdAt: z.string().datetime(),
});

export type SpecVersion = z.infer<typeof SpecVersionSchema>;

export const SpecSchema = z.object({
  id: z.string().uuid(),
  nodeId: z.string().uuid(),
  title: z.string().min(1),
  status: SpecStatus,
  sourceTaskId: z.string().uuid().nullable(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
  versions: z.array(SpecVersionSchema).optional(),
});

export type Spec = z.infer<typeof SpecSchema>;

File: src/backend/AgenticCompany.Core/Entities/Spec.cs
using AgenticCompany.Core.Enums;

namespace AgenticCompany.Core.Entities;

public class Spec
{
    public Guid Id { get; set; }
    public Guid NodeId { get; set; }
    public string Title { get; set; } = string.Empty;
    public SpecStatus Status { get; set; } = SpecStatus.Draft;
    public Guid? SourceTaskId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Node Node { get; set; } = null!;
    public TaskItem? SourceTask { get; set; }
    public ICollection<SpecVersion> Versions { get; set; } = [];
    public ICollection<Plan> Plans { get; set; } = [];
}
